app/frontend/courses/students/types/CoursesStudents__StudentDetails.res

```diff
@@ -13,6 +13,7 @@ type t = {
   id: string,
   coachNotes: array<CoursesStudents__CoachNote.t>,
   hasArchivedNotes: bool,
+  preview: bool,
```

Comment:
Why not use `reviewable`; We should stick to `reviewable` or `preview`; Else it would create confusion over time;

Reply:
I also think `reviewable` works better in this context. Preview = _opportunity to view something before it is acquired or becomes generally available_.

_Reviewable_ isn't the best either. I believe this is being used in the context of the student report... where no _review_ will be performed, but it's better than _preview_. 🤷🏼‍♂️

Reply:
@harigopal A reviewed submission returning reviewable felt odd to me; Also that page has more actions than just review; (it feel like the check should have been on the lines of `hasWriteAccess`)

Reply:
@harigopal 

> Preview = opportunity to view something before it is acquired or becomes generally available.

I guess in our code, and UI we show a course curriculum as preview for admin/coach if they are not a student in that course.🤔 

It says preview mode 

Similarly, here, it same right? Admin doesn't have a coach profile so can't we call as preview like in the curriculum?

Reply:
> we show a course curriculum as preview for admin/coach if they are not a student in that course
﻿
@yash-learner The _preview_ mode for the course curriculum was actually named for the _public preview_ feature, which is where this was first implemented. It was a way for members of the public to preview a course before joining as a student. In that sense, it was a preview - the same _preview_ feature was later extended to allow admins and coaches to view the course without joining as a student.
﻿
This is just a question of using words that are appropriate for the context in which they're used. You could just as well call it _foo_, and it would _function_ just as well - it simply wouldn't communicate the intent of the boolean all that well.
﻿
_Preview_ is _okay_, but IMHO not the right word in either the student report page or in the submission review page.

Reply:
> it feels like the check should have been on the lines of hasWriteAccess

@bodhish `editable`, then? 🙂

Reply:
I went with `canModifyCoachNotes` in student report 😁

Reply:
`editable` would have been a good choice; `canModifyCoachNotes` is super context-specific and would work for the specific page and related action; My concern was more related to reviewable
> A reviewed submission returning `reviewable: true` felt odd to me; Also that page has more actions than just review; (it feels like the check should have been on the lines of hasWriteAccess)

Reply:
> A reviewed submission returning reviewable: true felt odd to me; Also that page has more actions than just review; (it feels like the check should have been on the lines of hasWriteAccess)

I checked with people who usually review submissions, All actions such as grading, adding feedback, undoing a grading and then adding a new feedback all comes under reviewing.

---

>  reviewed submission returning reviewable: true felt odd to me

A reviewed submission can still be reviewed, right? 🤔. By undoing a grade and adding a new one or adding a new feedback.

Reviewable: false will happen when a submission is from an ended cohort or dropped out. and when admin views it without being a coach.

I think all other actions are the subset of the review process.